### PR TITLE
Feature/use less reflection

### PR DESCRIPTION
**Pull request by john7doe**
Use less reflection. On android O it particularly slow to call getMethod, so make sure we are caching the results

### CODE REVIEW

File: server/instrumentation-backend/src/sh/calaba/instrumentationbackend/query/ast/UIQueryASTWith.java [hunk 1]
```diff
@@ -259,22 +286,19 @@ private boolean isMarked(Object o, Object expectedValue) {
...
+		List<Method> methods = textMethodsForClass.get(view.getClass());
+		for (Method	method : methods) {
 			try {
-				Method getTextM = view.getClass().getMethod(methodName);
-				Object text = getTextM.invoke(view);
+				Object text = method.invoke(view);
 				if (text != null && text.toString().equals(expected)) {
 					return true;
 				}
-			} catch (Exception e) {
```
Comment (owenniblock) [inline server/instrumentation-backend/src/sh/calaba/instrumentationbackend/query/ast/UIQueryASTWith.java]: Was there a reason for no longer doing a blanket `Exception` catch here? (Apart from the fact that it's better practice to catch specific Exceptions 🙂 )

**Comment by owenniblock** [comment]
Would be good to add a before and after test run on the PR for future reference 🙂

File: server/instrumentation-backend/src/sh/calaba/instrumentationbackend/query/ast/UIQueryASTWith.java (hunk 1 above)
Comment (john7doe) [reply server/instrumentation-backend/src/sh/calaba/instrumentationbackend/query/ast/UIQueryASTWith.java]: The two exceptions I catch are expected exceptions, anything else I would define as an coding error and I do not silently mute that.

**Comment by john7doe** [comment]
I used the Android profiler that's part of android studio. It's far from perfect, e.g. you cannot save profiler results. But with a two phones attached and two projects I was able to compare traces from 7.X and 8.X.

The class definitions will not change over time, so we can cache the getMethod() results.

Pre change:
https://testcloud.xamarin.com/test/webuyanycar_fa6fbb35-7b13-49c8-8e34-8d2e6ebffc61/

Post change:
https://testcloud.xamarin.com/test/41367fe3-7861-494d-ae4f-5a226e7460a2/

**Pull request merged**